assets/js/components/settings/SettingsActiveModule/Footer.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * Footer component for SettingsActiveModule.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Fragment, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_MODULES } from '../../../googlesitekit/modules/datastore/constants';
+import { Cell, Grid, Row } from '../../../material-components';
+import PencilIcon from '../../../../svg/pencil.svg';
+import TrashIcon from '../../../../svg/trash.svg';
+import Button from '../../Button';
+import Spinner from '../../Spinner';
+import Link from '../../Link';
+const { useSelect } = Data;
+
+export default function Footer( props ) {
+	const {
+		slug,
+		isSaving,
+		isEditing,
+		onConfirm,
+		onCancel,
+		onEdit,
+		handleDialog,
+	} = props;
+
+	const canSubmitChanges = useSelect( ( select ) => select( CORE_MODULES ).canSubmitChanges( slug ) );
+	const module = useSelect( ( select ) => select( CORE_MODULES ).getModule( slug ) );
+
+	const setupComplete = module?.connected;
+	const hasSettings = !! module?.SettingsEditComponent;
+
+	const handleConfirmOrCancel = useCallback( () => {
+		if ( hasSettings && setupComplete ) {
+			onConfirm();
+		} else {
+			onCancel();
+		}
+	}, [ hasSettings, setupComplete ] );
+
+	if ( ! module ) {
+		return null;
+	}
+
+	const {
+		name,
+		homepage,
+		forceActive: autoActivate,
+	} = module;
+
+	let primaryColumn = null;
+	let secondaryColumn = null;
+
+	if ( isEditing || isSaving ) {
+		let buttonText = __( 'Close', 'google-site-kit' );
+		if ( hasSettings && setupComplete ) {
+			buttonText = isSaving
+				? __( 'Saving…', 'google-site-kit' )
+				: __( 'Confirm Changes', 'google-site-kit' );
+		}
+
+		primaryColumn = (
+			<Fragment>
+				<Button
+					id={ hasSettings && setupComplete ? `confirm-changes-${ slug }` : `close-${ slug }` }
```

Comment:
I think the `id` should no longer be necessary now. It was only really used by `toggleConfirmModuleSettings` which is unused now anyways and should be removed (there's another issue for this but I think we can clean it up here if its target no longer exists).

Reply:
Removed.